### PR TITLE
Tests to make sure compiler produces errors on empty key path for signing

### LINKED ISSUE #11497: Inconsistent compiler behaviour between OS X and Windows if keyFile is the empty string
**Version Used**: 

_OS X_:

``` sh
computer:api user$ dotnet --info
.NET Command Line Tools (1.0.0-preview1-002702)

Product Information:
 Version:     1.0.0-preview1-002702
 Commit Sha:  6cde21225e

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  10.11
 OS Platform: Darwin
 RID:         osx.10.11-x64
```

_Windows_:

``` cmd
PS C:\Coding\api> dotnet --info
.NET Command Line Tools (1.0.0-preview1-002702)

Product Information:
 Version:     1.0.0-preview1-002702
 Commit Sha:  6cde21225e

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.10240
 OS Platform: Windows
 RID:         win10-x64
```

**Steps to Reproduce**:
1. Clone [https://github.com/martincostello/api.git](https://github.com/martincostello/api.git) from commit [c7d2e8a448470c07d98160bc19cbc2786cb28fc6](https://github.com/martincostello/api/commit/c7d2e8a448470c07d98160bc19cbc2786cb28fc6).
2. Run `dotnet build src/API` from the root of the repository.

**Expected Behaviour**:

OS X and Windows build attempts exhibit the same behaviour, which is either to:
1. Treat `""` as `null` on both platforms and compile successfully with no strong name, or;
2. Treat `""` as an invalid key file and error with `error CS7088: Invalid 'CryptoKeyFile' value: ''.` on both platforms. I'm tending towards this being the preferrable behaviour as in my case I didn't have a key file at all, so I resolved the error by using `null` explicitly in [this commit](https://github.com/martincostello/api/commit/0dc2557083103f2ac500ae871a9a901cb0c8a293).

**Actual Behaviour**:

_OS X_:

``` sh
computer:api user$ dotnet build src/API
Project API (.NETCoreApp,Version=v1.0) will be compiled because expected inputs are missing
Compiling API for .NETCoreApp,Version=v1.0
/usr/local/share/dotnet/dotnet compile-csc @/Users/user/Coding/api/src/API/obj/Debug/netcoreapp1.0/dotnet-compile.rsp returned Exit Code 1
/Users/user/Coding/api/src/API/error CS7088: Invalid 'CryptoKeyFile' value: ''.

Compilation failed.
    0 Warning(s)
    1 Error(s)

Time elapsed 00:00:01.5841885
```

_Windows_:

``` cmd
PS C:\Coding\api> dotnet build .\src\API\
Project API (.NETCoreApp,Version=v1.0) will be compiled because expected inputs are missing
Compiling API for .NETCoreApp,Version=v1.0

Compilation succeeded.
    0 Warning(s)
    0 Error(s)

Time elapsed 00:00:02.5016735
```

### PR DESCRIPTION
**Pull request by OmarTawfik**
Closes #11497 by adding appropriate tests.

### CODE REVIEW

**Comment by OmarTawfik** [comment]
@agocke From Jenkins, I see the new tests only running on desktop. Can you please tell me what attributes or other changes I should do to make them run everywhere?

File: src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs [hunk 1]
```diff
@@ -4367,6 +4367,15 @@ public void PublicSign_KeyFileRelativePath()
             Assert.Equal(Path.Combine(_baseDirectory, "test.snk"), parsedArgs.CompilationOptions.CryptoKeyFile);
         }
 
+        [Fact]
+        [WorkItem(11497, "https://github.com/dotnet/roslyn/issues/11497")]
+        public void PublicSignWithEmptyKeyPath()
+        {
+            DefaultParse(new[] { "/publicsign", "/keyfile:", "a.cs" }, _baseDirectory).Errors.Verify(
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs]: >"/keyfile:" [](start = 48, length = 11)

It feels like we should also test ```"/keyfile:"""""``` scenario.

**Comment by AlekseyTs** [comment]
Should we test similar scenarios for VB?

**Comment by AlekseyTs** [comment]
Done with review pass (iteration 1)

**Comment by OmarTawfik** [comment]
@AlekseyTs comments addressed.
@dotnet/roslyn-compiler  can I get a second review?

File: src/Compilers/CSharp/Test/Symbol/Compilation/CompilationAPITests.cs [hunk 2]
```diff
@@ -45,6 +45,26 @@ public void PublicSignWithRelativeKeyPath()
...
+                // error CS8102: Public signing was specified and requires a public key, but no public key was specified.
+                Diagnostic(ErrorCode.ERR_PublicSignButNoKey).WithLocation(1, 1));
+        }
+
+        [Fact]
+        [WorkItem(11497, "https://github.com/dotnet/roslyn/issues/11497")]
+        public void PublicSignWithEmptyKeyPath2()
+        {
+            CreateStandardCompilation("", options: TestOptions.ReleaseDll.WithPublicSign(true).WithCryptoKeyFile("\"\"")).VerifyDiagnostics(
+                // error CS8106: Option 'CryptoKeyFile' must be an absolute path.
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/Symbol/Compilation/CompilationAPITests.cs]: This error seems redundant (and also in VB). Is it easy to remove the cascade?

Comment (OmarTawfik) [reply src/Compilers/CSharp/Test/Symbol/Compilation/CompilationAPITests.cs]: We can. But that requires a minor API change (`AssemblyIdentity.HasPublicKey`), or considering empty strings as existing paths. I'm not sure if it was worth it to make that change.

---
In reply to: [156213571](https://github.com/dotnet/roslyn/pull/23646#discussion_r156213571) [](ancestors = 156213571)

Comment (jcouv) [reply src/Compilers/CSharp/Test/Symbol/Compilation/CompilationAPITests.cs]: Ok. Makes sense to leave then.

**Comment by jcouv** [comment]
@OmarTawfik Please fill in the template before submitting to MeiChin.

**Comment by OmarTawfik** [comment]
@AlekseyTs any other comments?

**Comment by OmarTawfik** [comment]
@jcouv this is a test only change. We don't need to ask for that?

**Comment by jcouv** [comment]
Ah, right. No need to ask-mode approval :-)

**Pull request merged**